lib/puppet/metatype/manager.rb

```diff
@@ -146,16 +156,19 @@ def type(name)
     # We are overwhelmingly symbols here, which usually match, so it is worth
     # having this special-case to return quickly.  Like, 25K symbols vs. 300
     # strings in this method. --daniel 2012-07-17
-    return @types[name] if @types[name]
+    return @types[name] if @types.include? name
```

Comment:
This will do a linear scan which could be bad if there are many types. Might be better to store a sentinel value in the case of a miss, e.g. `@types[name] = CacheMiss`, and then check for that value here, e.g. `return @types[name] if @types[name] != CacheMiss`